main.go

```diff
@@ -46,22 +50,10 @@ func main() {
 	command := flag.Arg(0)
 	switch {
 	case command == "update":
-		userEnv := os.Getenv("GITOR_USER")
-		if username == "" {
-			if userEnv == "" {
-				log.Fatal("username not set")
-			}
-			username = userEnv
-		}
-
-		tokenEnv := os.Getenv("GITOR_TOKEN")
-		if token == "" {
-			if tokenEnv == "" {
-				log.Fatal("token or password not set")
-			}
-			token = tokenEnv
-		}
-		update(upstream, branch, username, token, downstream)
+		username, token := checkEnvs(username, token)
+		upstreamAuth := authType(upstream, username, token)
+		downstreamAuth := authType(upstream, username, token)
+		update(upstream, branch, downstream, upstreamAuth, downstreamAuth)
```

Comment:
Things still missing:
* split update into pull and push aka upstream and downstream
* branch not settable on downstream yet